pad.go

```diff
@@ -80,20 +81,23 @@ func (pad *PAD) Update(policies Policies) {
 }
 
 func (pad *PAD) Set(key string, value []byte) error {
-	return pad.tree.Set(key, value)
+	index := pad.computePrivateIndex(key)
+	return pad.tree.Set(index, key, value)
 }
 
 func (pad *PAD) Lookup(key string) *AuthenticationPath {
 	str := pad.currentSTR
-	return str.tree.Get(key)
+	lookupIndex := pad.computePrivateIndex(key)
+	return str.tree.Get(lookupIndex)
 }
 
 func (pad *PAD) LookupInEpoch(key string, epoch uint64) (*AuthenticationPath, error) {
 	str := pad.GetSTR(epoch)
 	if str == nil {
 		return nil, ErrorSTRNotFound
 	}
-	ap := str.tree.Get(key)
+	lookupIndex := pad.computePrivateIndex(key)
```

Comment:
I'm not sure what you mean. Maybe the `key` variable name is a little misleading here? The (private-)key which is used to compute the priv. is this one: https://github.com/coniks-sys/libmerkleprefixtree-go/pull/19/files#diff-73fecc7207210fbe4ae4c1365f307927R112